packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -100,8 +100,11 @@ class FlutterDevice {
     if (flutterViews == null || flutterViews.isEmpty)
       return;
     for (FlutterView view in flutterViews) {
-      if (view != null && view.uiIsolate != null)
+      if (view != null && view.uiIsolate != null) {
+        // Manage waits specifically below.
```

Comment:
Thanks @zanderso. I suspect that I copied that code as this is interacting the flutter app exit service protocol extension I know nothing about.